vermouth/tests/test_ff_files.py

```diff
@@ -0,0 +1,1019 @@
+        block = ff.blocks['GLY']
+
+        bonds = [
+            vermouth.molecule.Interaction(
+                atoms=['BB', 'SC1'], parameters=['1', '0.2', '100'], meta={},
+            ),
+            vermouth.molecule.Interaction(
+                atoms=['SC1', 'SC2'], parameters=['4', '0.6', '700'],
+                meta={'comment': 'custom'},
+            ),
+            vermouth.molecule.Interaction(
+                atoms=['SC2', 'BB'], parameters=['9', 'plop', 'toto'], meta={},
+            ),
+        ]
+        assert block.interactions['bonds'] == bonds
+
+    # test_interaction_edges(interaction_lines, edges)
+    @staticmethod
+    @pytest.mark.parametrize(
+        'interaction_lines, edges', (
+            (  # regular bonds create edges
+                """
+                [ bonds ]
+                BB SC1
+                SC2 SC3
+                """,
+                (('BB', 'SC1'), ('SC2', 'SC3')),
+            ),
+            (  # regular angles create edges
+                """
+                [ angles ]
+                BB SC1 SC2
+                SC1 SC2 SC3
+                """,
+                (('BB', 'SC1'), ('SC1', 'SC2'), ('SC2', 'SC3')),
+            ),
+            (  # regular dihedrals create edges
+                """
+                [dihedrals]
+                BB SC1 SC2 SC3
+                """,
+                (('BB', 'SC1'), ('SC1', 'SC2'), ('SC2', 'SC3')),
+            ),
+            (  # impropers do not create edges
+                """
+                [ impropers ]
+                BB SC1 SC2 SC3
+                """,
+                (),
+            ),
+            (
+                """
+                [ dihedrals ]
+                BB SC1 SC2 SC3 2 ; gromacs dihedrals of type 2 are impropers
+                """,
+                (),
+            ),
+            (  # the edge section creates edges
+                """
+                [ edges ]
+                SC1 SC2
+                BB SC3
+                """,
+                (('SC1', 'SC2'), ('BB', 'SC3')),
+            ),
+            (  # edges can be deactivated from interactions
+                """
+                [ bonds ]
+                BB SC1 -- {"edge": false}
+                SC2 SC3 -- {"edge": true}
+                """,
+                (('SC2', 'SC3'), ),
+            ),
+    ))
+    def test_interaction_edges(interaction_lines, edges):
+        """
+        Edges are created where expected.
+        """
+        lines = """
+        [ moleculetype ]
+        GLY 1
+
+        [ atoms ]
+        1 P4 1 ALA BB 1
+        2 P3 1 ALA SC1 2
+        3 P2 1 ALA SC2 3
+        4 P2 1 ALA SC3 3
+        """
+        lines = textwrap.dedent(lines) + textwrap.dedent(interaction_lines)
+        lines = lines.splitlines()
+
+        ff = vermouth.forcefield.ForceField(name='test_ff')
+        vermouth.ffinput.read_ff(lines, ff)
+        block = ff.blocks['GLY']
+
+        assert set(block.edges) == set(edges)
```

Comment:
Edges are symmetric. In this case though, `{("A", "B"),} != {("B", "A")}`

Reply:
Yes, and it breaks the tests on python 3.4 and 3.5 :cry: I need to compare sets of frozensets.